yt_dlp/extractor/piaulizaportal.py

```diff
@@ -1,11 +1,61 @@
 from .common import InfoExtractor
-from ..utils import (
-    ExtractorError,
-    int_or_none,
-    parse_qs,
-    time_seconds,
-    traverse_obj,
-)
+from ..utils import ExtractorError, int_or_none, parse_qs, time_seconds
+from ..utils.traversal import traverse_obj
+
+
+class PIAULIZAPortalAPIIE(InfoExtractor):
+    _VALID_URL = r'https://player-api\.p\.uliza\.jp/v1/players/[^?#]+\?(?:[^#]*&)?name=(?P<id>[^#&]+)'
+    _TESTS = [
+        {
+            'url': 'https://player-api.p.uliza.jp/v1/players/timeshift-disabled/pia/admin?type=normal&playerobjectname=ulizaPlayer&name=livestream01_dvr&repeatable=true',
+            'info_dict': {
+                'id': '88f3109a-f503-4d0f-a9f7-9f39ac745d84',
+                'title': '88f3109a-f503-4d0f-a9f7-9f39ac745d84',
+                'live_status': 'was_live',
+            },
+        },
+        {
+            'url': 'https://player-api.p.uliza.jp/v1/players/uliza_jp_gallery_normal/promotion/admin?type=presentation&name=cookings&targetid=player1',
+            'info_dict': {
+                'id': 'ae350126-5e22-4a7f-a8ac-8d0fd448b800',
+                'title': 'ae350126-5e22-4a7f-a8ac-8d0fd448b800',
+                'live_status': 'not_live',
+            },
+        },
+        {
+            'url': 'https://player-api.p.uliza.jp/v1/players/default-player/pia/admin?type=normal&name=pia_movie_uliza_fix&targetid=ulizahtml5&repeatable=true',
+            'info_dict': {
+                'id': '0644ecc8-e354-41b4-b957-3b08a2d63df1',
+                'title': '0644ecc8-e354-41b4-b957-3b08a2d63df1',
+                'live_status': 'not_live',
+            },
+        },
+    ]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        player_data = self._download_webpage(
+            url, display_id, headers={'Referer': 'https://player-api.p.uliza.jp/'},
+            note='Fetching player data', errnote='Unable to fetch player data',
+        )
+
+        m3u8_url = self._search_regex(
+            r'["\'](https://vms-api\.p\.uliza\.jp/v1/prog-index\.m3u8[^"\']+)', player_data,
+            'm3u8 url', default=None)
+        video_id = self._search_regex(r'&?ss=([\da-f]{8}-(?:[\da-f]{4}-){3}[\da-f]{12})&?', m3u8_url, 'video id', default=display_id)
```

Comment:
Since that's a URL:

```suggestion
        video_id = parse_qs(m3u8_url).get('ss', [display_id])[0]
```

Don't forget `from ..utils import parse_qs`.

Reply:
Changed by a75a02ad2a96423f3bf87ba7310b3603bf00cb85.